runtime/doc/lua.txt

```diff
@@ -733,17 +733,222 @@ VIM.JSON                                                            *lua-json*
 
 The *vim.json* module provides encoding and decoding of Lua objects to and
 from JSON-encoded strings. Supports |vim.NIL| and |vim.empty_dict()|.
+This module is provided by `openresty/lua-cjson`. See
+https://github.com/openresty/lua-cjson for more documentation.
 
-vim.json.encode({obj})                                       *vim.json.encode*
+encode({obj})                                              *vim.json.encode()*
     Encodes (or "packs") Lua object {obj} as JSON in a Lua string.
 
-vim.json.decode({str}[, {opts}])                             *vim.json.decode*
+decode({str}[, {opts}])                                    *vim.json.decode()*
     Decodes (or "unpacks") the JSON-encoded {str} to a Lua object.
 
     {opts} is a table with the key `luanil = { object: bool, array: bool }`
     that controls whether `null` in JSON objects or arrays should be converted
     to Lua `nil` instead of `vim.NIL`.
 
+new()                                                         *vim.json.new()*
+    Creates and returns a new `json` object, which is seperate from the
+    global `vim.json`. Can be used to safely configure the options on your
+    object and ensure safe parsing and encoding.
+
+The following methods change the configuration of the json object. ~
+The current settings are always returned. A particular setting is only ~
+changed when the argument is provided (non-nil). ~
+
+NOTE: These are global changes! Be careful with them or else create a new
+object with |vim.json.new()|
+
+                                     *vim.json.encode_empty_table_as_object()*
+encode_empty_table_as_object([{enable}])
+    Default: `false`
+    Parameters: ~
+      • {enable} (boolean / `"on"` / `"off"`) Whether to include array_mt
+        metatable.
+
+    Return: ~
+      Current value: {enable}
+
+    Change the default behavior when encoding an empty Lua table.
+
+    `true` - empty Lua tables are encoded as empty JSON Objects (`{}`).
+    `false` - empty Lua tables are encoded as empty JSON Arrays (`[]`).
+
+   {enable} is either a boolean or a string (`"on"`, `"off"`).
+
+                                       *vim.json.decode_array_with_array_mt()*
+decode_array_with_array_mt({bool})
+    Default: `false`
+    Parameters: ~
+      • {bool} (boolean) Whether to include array_mt metatable.
+
+    Return: ~
+      Current value: {bool}
+    `true` - JSON Arrays decoded by |vim.json.decode| will result in Lua tables
```

Comment:
because you are missing a pair of parentheses here
```suggestion
    `true` - JSON Arrays decoded by |vim.json.decode()| will result in Lua tables
```